tedpermission/src/main/java/com/gun0912/tedpermission/PermissionBuilder.java

```diff
@@ -64,30 +79,63 @@ protected void checkPermissions() {
         intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
         intent.addFlags(Intent.FLAG_ACTIVITY_NO_USER_ACTION);
         TedPermissionActivity.startActivity(context, intent, listener);
-        TedPermissionBase.setFirstRequest(context,permissions);
+        TedPermissionBase.setFirstRequest(context, permissionsAsStringArray());
+    }
+
+    @NonNull
+    private String[] permissionsAsStringArray() {
+        return permissions.toArray(new String[permissions.size()]);
     }
 
     public T setPermissionListener(PermissionListener listener) {
-        this.listener = listener;
+        if (listener != null) {
+            this.listener = listener;
+        }
         return (T) this;
     }
 
     public T setPermissions(String... permissions) {
-        this.permissions = permissions;
+        if (permissions != null) {
+            this.permissions.clear();
+            this.permissions.addAll(Arrays.asList(permissions));
+        }
         return (T) this;
     }
 
-    public T setRationaleMessage(@StringRes int stringRes) {
-        return setRationaleMessage(getText(stringRes));
+    public T addPermissions(List<String> permissions) {
+        if (permissions != null) {
+            this.permissions.addAll(permissions);
+        }
+        return (T) this;
+    }
+
+    public T addSinglePermission(String permission) {
```

Comment:
how about change naming to `addPermission`?
(It's just a recommendation. This is also good.)

Reply:
Thanks for your comment. I thought about it but I guess `addPermissions` and `addPermission` look too similar and it would be easy to mix them up.

Reply:
You're right. Thanks